File: solutions/179.largest-number.js
/*
 * @lc app=leetcode id=179 lang=javascript
 *
 * [179] Largest Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function (nums) {
  const res = nums
    .map(String)
    .sort((a, b) => {
      if (a + b === b + a) return 0;
      return a + b < b + a ? 1 : -1;
    })
    .join("");
  return res[0] === "0" ? "0" : res;
};
// @lc code=end

File: solutions/179.largest-number.rb
# @param {Integer[]} nums
# @return {String}
def largest_number(nums)
  res = nums.map(&:to_s).sort{
    if _1 + _2 == _2 + _1 then 0
    else _1 + _2 < _2 + _1 ? 1 : -1 end
   }.join
  res[0] == '0' ? '0' : res
end
